packages/manager/src/features/linodes/LinodesDetail/LinodeNetworking/LinodeNetworking.tsx

```diff
@@ -229,10 +244,69 @@ class LinodeNetworking extends React.Component<CombinedProps, State> {
           initialLoading: false,
         });
       });
+
+    const refreshIPv6 = getAllIPv6Ranges({})
+      .then(async (resp) => {
+        const ranges = resp.data;
+
+        const sharedRanges: IPRange[] = [];
+        const staticRanges: IPRange[] = [];
+        const availableRanges: IPRangeInformation[] = [];
+        const rangeConstruction = await ranges.reduce(async (acc, range) => {
+          await acc;
+          // filter user ranges outside dc
+          if (range.region !== this.props.linode.region) {
+            return acc;
+          }
+
+          await getIPv6RangeInfo(range.range).then(async (resp) => {
+            let slaac;
+            if (!resp.is_bgp) {
+              await getLinodeIPs(this.props.linode.id).then((ips) => {
+                slaac = ips?.ipv6?.slaac.address;
+              });
+            }
+
+            // any range that is shared to this linode
+            if (resp.is_bgp && resp.linodes.includes(this.props.linode.id)) {
+              this.props.flags.ipv6Sharing && sharedRanges.push(range);
```

Comment:
I don't know how this should work, but to me, its confusing that when a Linode that "owns" an IPv6 range, if the IP is shared, in the Share IP modal, the IP shows up there... Should the shared range only show on the Linode's modal that doesn't "own the IP"... 

The code below explains what I mean, but needs modification in `ipResponseToDisplayRows` to work as I'm explainig

```suggestion
            if (
              resp.is_bgp &&
              resp.linodes.includes(this.props.linode.id) &&
              !this.state.linodeIPs?.ipv6?.global.some(
                (thisRange) => thisRange.range === range.range
              )
            ) {
              this.props.flags.ipv6Sharing && sharedRanges.push(range);
```

Reply:
It's a little strange (took me a long time to get), but this is the intended behavior. Showing it allows us to unshare that Linode from that range.

We want ranges not shared to a Linode in availableRanges, and ranges that are shared to a Linode in shared ranges